task_sdk/src/airflow/sdk/api/client.py

```diff
@@ -169,6 +169,19 @@ def get(self, dag_id: str, run_id: str, task_id: str, key: str, map_index: int =
         resp = self.client.get(f"xcoms/{dag_id}/{run_id}/{task_id}/{key}", params={"map_index": map_index})
         return XComResponse.model_validate_json(resp.read())
 
+    def set(
+        self, dag_id: str, run_id: str, task_id: str, key: str, value, map_index: int | None = None
+    ) -> dict[str, bool]:
+        """Set a XCom value via the API server."""
+        params = {}
+        if map_index:
+            params = {"map_index": map_index}
+        self.client.post(f"xcoms/{dag_id}/{run_id}/{task_id}/{key}", params=params, json=value)
```

Comment:
I'm not sure map_index should be a query param. I think in Airflow API when a task is mapped we add an extra `f"/{map_index}` part to the URL. Does that sound right/familiar to you @uranusjr @jedcunningham or am I making stuff up again?

Reply:
Tried with map_index as a query param and it works:
![image](https://github.com/user-attachments/assets/3b3e1afe-b936-41a7-8fea-7bc0feb42c6d)

Even its usage in https://github.com/apache/airflow/blob/main/airflow/api_fastapi/core_api/routes/public/xcom.py#L57 hints that it is a query param.

Even tests for that matter:
```
        response = test_client.get(
            "/public/dags/~/dagRuns/~/taskInstances/~/xcomEntries",
            params={"map_index": map_index} if map_index is not None else None,
        )
```

Reply:
Should I leave a comment on this one and we can check again later?